modules/cudf/src/data_frame.ts

```diff
@@ -1587,4 +1587,75 @@ export class DataFrame<T extends TypeMap = any> {
       this.names.reduce((cols, name) => ({...cols, [name]: this.get(name).isNotNull()}),
                         {} as SeriesMap<{[P in keyof T]: Bool8}>));
   }
+
+  /**
+   * Replace null values with a scalar value.
+   *
+   * @param value The scalar value to use in place of nulls.
+   * @param memoryResource The optional MemoryResource used to allocate the result Column's device
+   *   memory.
+   *
+   * @example
+   * ```typescript
+   * import {DataFrame, Series} from '@rapidsai/cudf';
+   *
+   * const df = new DataFrame({
+   *  a: Series.new([0, null, 2]);
+   *  b: Series.new([null, null, null]);
+   * });
+   *
+   * df.replaceNulls(1);
+   * // return {
+   * //    a: [0, 1, 2],
+   * //    b: [1, 1, 1],
+   * // }
+   * ```
+   */
+  replaceNulls(value: T['scalarType']|any, memoryResource?: MemoryResource): DataFrame;
+
+  /**
+   * Replace null values with the corresponding elements from another Map of Series.
+   *
+   * @param value The map of Series to use in place of nulls.
+   * @param memoryResource The optional MemoryResource used to allocate the result Column's device
+   *   memory.
+   *
+   * @example
+   * ```typescript
+   * import {DataFrame, Series} from '@rapidsai/cudf';
+   *
+   * const df = new DataFrame({
+   *  a: Series.new([0, null, 2]);
+   *  b: Series.new([null, null, null]);
+   * });
+   *
+   * df.replaceNulls({'a': Series.new([0, 1, 2]), 'b': Series.new([1, 1, 1])});
+   * // return {
+   * //    a: [0, 1, 2],
+   * //    b: [1, 1, 1],
+   * // }
+   * ```
+   */
+  replaceNulls(value: SeriesMap<T>, memoryResource?: MemoryResource): DataFrame;
+
+  replaceNulls(value: any, memoryResource?: MemoryResource): DataFrame {
+    if (value instanceof Object) {
```

Comment:
since `SeriesMap` is an interface, I'm unable to do a check against `instanceof`, any other suggestions?

Reply:
I'm not sure about the type check, but I think the signature can be improved?
```
  replaceNulls<R extends DataType>(value: R['scalarType']|SeriesMap<T>,
                                   memoryResource?: MemoryResource): DataFrame<T> {
```

For checking against interfaces folks seem to variously suggest adding s type tag field to the object but I don't think we can do that safely since the names in the map can be arbitrary and might conflict with anything we chose. Maybe there is some way to check or the converse, i.e. is a scalar type? Not sure. I do think with the narrower signature the check against object is a little less worrisome, with the possible exception of struct columns? Their scalar type is an object too `scalarType: {[P in keyof T]: T[P]['scalarType']};`

Reply:
I'll see if the converse can be done,

Also sorry this comment is referencing an older version of this PR, this is how it currently looks...
```
replaceNulls(value: T['scalarType']|any, memoryResource?: MemoryResource): DataFrame;
```

Should be basically the same as you've suggested above (I'll double check this).

Reply:
ah yes, sorry for not checking the current diff!

Reply:
Hmm, no luck. There isn't really a neat way to check against interfaces like you've suggested.

We can type guard the generic `any` method though...

```
replaceNulls<R extends DataType>(value: SeriesMap<T>|R['scalarType'],
                                 memoryResource?: MemoryResource): DataFrame<T> {
```